ui/components/app/signature-request/signature-request.component.js

```diff
@@ -28,33 +25,6 @@ export default class SignatureRequest extends PureComponent {
     metricsEvent: PropTypes.func,
   };
 
-  componentDidMount() {
-    if (getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
-      window.addEventListener('beforeunload', this._beforeUnload);
-    }
-  }
-
-  _beforeUnload = (event) => {
-    const {
-      clearConfirmTransaction,
-      cancel,
-      txData: { type },
-    } = this.props;
-    const { metricsEvent } = this.context;
-    metricsEvent({
-      eventOpts: {
-        category: 'Transactions',
-        action: 'Sign Request',
-        name: 'Cancel Sig Request Via Notification Close',
-      },
-      customVariables: {
```

Comment:
Note this extra data  here. This component (`signature-request.component.js`) is used for `eth_signTypedData_v3` and `eth_signTypedData_v4`). So this data should be added in the `typedMessageManager` specifically for types V3 and V4, and not for type V1.

Though honestly the extra data couldn't hurt :thinking: Maybe we can add it for all types. I highly doubt any report is counting on it _not_ being set. These metric events are likely to be replaced with more sensible ones in the not too distant future anyway.

Reply:
It looks like this custom variable is still missing from the `typedMessageManager` metric event. We need it to be present at least for v3 and v4 messages, because this is how we distinguish between them on our metrics dashboard.

Reply:
Oh, it looks like I was wrong. This `type` doesn't represent the version at all. It just says that it's a signed type message :disappointed: . Oh well. Sorry about that mixup. I see that you've added it to the metric events already by now though, and that seems useful regardless.